src/java.rmi/share/classes/sun/rmi/transport/tcp/TCPTransport.java

```diff
@@ -285,7 +290,7 @@ private void decrementExportCount() {
             tcpLog.log(Log.VERBOSE,
                     "server socket: " + server + ", exportCount: " + exportCount);
         }
-        if (exportCount == 0 && getEndpoint().getListenPort() != 0) {
+        if (exportCount == 0) {
```

Comment:
In my opinion, this is the most risky change in this PR; I am not sure what was the motivation for not closing the server socket listening on the random port.